src/coprocessor/dag/scanner.rs

```diff
@@ -142,11 +127,6 @@ impl<S: Store> Scanner<S> {
             util::convert_to_prefix_next(&mut self.last_scanned_key);
             range.set_end(self.last_scanned_key.clone());
         } else {
-            // TODO: We may don't need `truncate_as_row_key`. Needs investigation.
-            if self.scan_on == ScanOn::Table {
-                let row_key_len = truncate_as_row_key(&self.last_scanned_key).unwrap().len();
-                self.last_scanned_key.truncate(row_key_len);
-            }
```

Comment:
Why we don't need `truncate_as_row_key`?

Reply:
Because the key itself is the row key. There is nothing extra.